src/freertos_drivers/esp_idf/EspIdfWiFi.cxx

```diff
@@ -0,0 +1,1355 @@
+            nvs_set_blob(cfg_, NVS_KEY_LAST_NAME, &userCfg_, sizeof(userCfg_));
+            nvs_commit(cfg_);
+            break;
+        default:
+            LOG(LEVEL_ERROR, "wifi: Error %s getting privCfg_.",
+                esp_err_to_name(result));
+            break;
+    }
+}
+
+//
+// EspIdfWiFi::init_wifi()
+//
+void EspIdfWiFi::init_wifi(WlanRole role)
+{
+    if (role == WlanRole::DEFAULT_ROLE)
+    {
+        role = userCfg_.mode_;
+    }
+
+    ESP_ERROR_CHECK(esp_netif_init());
+
+    /// @todo Do we need this?
+    esp_err_t result = esp_event_loop_create_default();
+    if (result != ESP_OK)
+    {
+        LOG(LEVEL_ERROR, "wifi: esp_event_loop_create_default() failed: %s",
+            esp_err_to_name(result));
+    }
+
+    ESP_ERROR_CHECK(esp_event_handler_instance_register(
+        WIFI_EVENT, ESP_EVENT_ANY_ID, &wifi_event_handler, this, nullptr));
+    ESP_ERROR_CHECK(esp_event_handler_instance_register(
+        IP_EVENT, ESP_EVENT_ANY_ID, &ip_event_handler, this, nullptr));
+
+    // Use this to debug wifi connection problems
+    // esp_log_level_set("wifi", ESP_LOG_VERBOSE);
+
+    wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
+    ESP_ERROR_CHECK(esp_wifi_init(&cfg));
+
+
+    // Set and initialize the appropriate role(s).
+    switch (role)
+    {
+        default:
+            // Fall through.
+        case WlanRole::UNKNOWN:
+            // Fall through.
+        case WlanRole::AP_STA:
+            ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_APSTA));
+            init_softap(userCfg_.ap_.ssid_, userCfg_.ap_.pass_);
+            init_sta();
+            break;
+        case WlanRole::AP:
+            ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_AP));
+            init_softap(userCfg_.ap_.ssid_, userCfg_.ap_.pass_);
+            break;
+        case WlanRole::STA:
+            ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_STA));
+            init_sta();
+            break;
+    }
+    // Start the WiFi.
+    ESP_ERROR_CHECK(esp_wifi_start());
+
+    if (apIface_)
+    {
+        esp_netif_ip_info_t ip_info;
+        esp_netif_get_ip_info(
+            esp_netif_get_handle_from_ifkey("WIFI_AP_DEF"), &ip_info);
+
+        char ip_addr[16];
+        inet_ntoa_r(ip_info.ip.addr, ip_addr, 16);
+        LOG(INFO, "wifi: Set up softAP with IP: %s", ip_addr);
+
+        LOG(INFO, "wifi: init_softap() finished. SSID:'%s' password:'%s'",
+            userCfg_.ap_.ssid_, userCfg_.ap_.pass_);
+    }
+
+    // Initialize mDNS.
+    ESP_ERROR_CHECK(mdns_init());
+    mdns_hostname_set(hostname_);
+    mdns_instance_name_set(hostname_);
+}
+
+//
+// EspIdfWiFi::init_softap()
+//
+void EspIdfWiFi::init_softap(std::string ssid, std::string pass)
+{
+    apIface_ = esp_netif_create_default_wifi_ap();
+    esp_netif_set_hostname(apIface_, hostname_);
+
+    wifi_config_t conf;
+    memset(&conf, 0, sizeof(wifi_config_t));
+
+    conf.ap.max_connection = 4;
+    conf.ap.beacon_interval = 100;
```

Comment:
IIRC this is the default interval but it should be moved to a constant instead of hard coding with no notes as to what scale this would be (it is milliseconds IIRC)

Reply:
A comment indicating it would be great, I'm not a fan of hard coded "magic" numbers with clear comments on them.

Reply:
Added.